### ISSUE TITLE
Future investigations into helm chart?

### ISSUE THREAD

**Issue by aegershman** [splunk/docker-splunk#177]
I realize kubernetes is not an _officially_ supported deployment target, but [there is an entire folder of `test_scenarios` dedicated to kubernetes deployments](https://github.com/splunk/docker-splunk/tree/develop/test_scenarios/kubernetes).

For deploying heavy forwarders, we've taken the docker-splunk Concourse image and packaged it into a basic helm chart & use that to deploy the HEC in various environments.

At some point, though, it'd be nice to get out of the business of managing a custom helm chart & use something that someone smarter than I has a vision for.

Should we expect a helm chart (even if not officially supported) at some point that can be used to deploy various topologies? Not to be presumptive or entitled, just trying to get a picture of what may be available in the future.

Thanks for your time

**Comment by lephino** [comment]
I'll leave this one for @mikedickey to talk more on.  I'll say this much though, we're investigating several deployment methods for kubernetes.  As a feedback item, how do you feel about operators vs helm charts vs native deployment descriptors?

**Comment by aegershman** [comment]
tl;dr, helm

- Operators are a bit mysterious to me (it feels very "fire-and-forget/hope-for-the-best", limited options for error-handling, unclear exactly what an operator is doing at any given point yet I've just handed it power to my cluster, they're all implemented as CRDs, etc.).

- Native deployment descriptors are fine when you're doing "pet deployments", but considering I have to stamp out multiple deployments on multiple clusters in multiple IaaSes in multiple regions in multiple environments... ehhh it's not so great. How do I check everything into version control? How do I manage the exact packaged version used in each env? How do I handle slight variations in parameter tuning when spanning IaaSes and envs? etc. You're still going to have to template and version-control everything, but now instead of it being templating logic encapsulated by the maintainer of the chart, it's something that I as the consumer have to handle-- and in most cases, I'm not the expert. I just want to modify a limited subset of parameters that you, the maintainer, feel should be modified. Helm isn't perfect, but it's currently the best way, in my opinion, to handle versioned deployments in a git-backed declarative manner.

- helm charts are, currently, in my opinion, the best option. For managing multi-deployments on multi-clusters, I'm experimenting with using [`helmfile`](https://github.com/roboll/helmfile) to manage everything. It's obviously not perfect but so far it's the best way to handle multiple clusters with multiple deployments, one of which I'd like to be a Splunk HEC for [forwarding logs on from PAS](https://github.com/cloudfoundry-community/splunk-firehose-nozzle) and shipping them to our enterprise Splunk receiver (deployed traditionally on VMs and such).

For me, the whole goal is to make standing up deployments as declarative as possible. I don't want to have to futz around with each deployment individually & document special upgrade steps, special "pre-deployment considerations", etc. As much as possible, I strive to organize my deployment config so that I could, theoretically, stand up an entirely new cluster, run `helmfile apply` (or something thereof), and have all deployments ready to rock. 

thoughts?

**Comment by mikedickey** [comment]
Published some of our thoughts on this topic yesterday in a blog article: https://www.splunk.com/blog/2019/05/08/an-insider-s-guide-to-splunk-on-containers-and-kubernetes.html
